src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx.cs

```diff
@@ -235,64 +235,23 @@ public static class Avx
         /// </summary>
         public static Vector256<float> DuplicateOddIndexed(Vector256<float> value) => DuplicateOddIndexed(value);
 
-        /// <summary>
-        /// __int8 _mm256_extract_epi8 (__m256i a, const int index)
-        ///   HELPER
-        /// </summary>
-        public static sbyte Extract(Vector256<sbyte> value, byte index)
-        {
-            unsafe
-            {
-                index &= 0x1F;
-                sbyte* buffer = stackalloc sbyte[32];
-                Store(buffer, value);
-                return buffer[index];
-            }
-        }
-
         /// <summary>
         /// __int8 _mm256_extract_epi8 (__m256i a, const int index)
         ///   HELPER
         /// </summary>
         public static byte Extract(Vector256<byte> value, byte index)
         {
-            unsafe
-            {
-                index &= 0x1F;
-                byte* buffer = stackalloc byte[32];
-                Store(buffer, value);
-                return buffer[index];   
-            }
+            return Unsafe.Add<byte>(ref Unsafe.As<Vector256<byte>, byte>(ref value), index & 0x1F);
```

Comment:
Removed `sbyte` and `short` overloads of SSE2/SSE4.1/AVX `Extract` and simplify `Avx.Extract` non-const fallback as @jkotas's suggestion.

I would prepare the CoreFX counterpart if this PR looks good to you guys.